### ISSUE TITLE
No weight decay for distillation token

### ISSUE THREAD

**Issue by bhheo** [facebookresearch/deit#48]
Hi

I have read distillation code and I think there is some error

As I know, `no_weight_decay` function of timm vision transformer doesn't covers the distillation token

https://github.com/rwightman/pytorch-image-models/blob/f8463b8fa9c0490db093b36acfce71fa2363b8c3/timm/models/vision_transformer.py#L254-L256

So, `no_weight_decay` function has to be overrided in `DistilledVisionTransformer`
```
@torch.jit.ignore
def no_weight_decay(self):
    return {'pos_embed', 'cls_token', 'dist_token'}
```

**Comment by TouvronHugo** [comment]
Hi @bhheo,
Thanks for your remark,
It is normal we use the weight decay with the distillation tokens.
Best,
Hugo

**Comment by bhheo** [comment]
Oh, I misunderstood it.
Thanks for the correction.
